### ISSUE TITLE
Error applying site template with field containing curling brackets

### ISSUE THREAD

**Issue by NerijusV** [SharePoint/PnP-Sites-Core#1045]
#### Category
[x] Bug
[ ] Enhancement

#### Environment
[x] Office 365 / SharePoint Online
[  ] SharePoint 2016
[x] SharePoint 2013

If SharePoint on-premises, what's exact CU version: 

#### Expected or Desired Behavior
Trying to apply PnP provisioning schema containing fields having curling bratkets `{` or `}` in display name or default value. Example:

```
<Field Type="Text" DisplayName="My {12} field" Required="TRUE" ...>
     <Default>{0}</Default>
</Field>
```

#### Observed Behavior
PnP site provisioning fails. Error text:

> An unhandled exception of type 'System.Exception' occurred in OfficeDevPnP.Core.dll
> 
> Additional information: The field was found invalid:  {0}

#### Steps to Reproduce
Try to apply provisioning template with field containing curling brackets in default value or field name.

**Comment by jansenbe** [comment]
Thanks for reporting this @NerijusV, we'll further investigate this

**Comment by luismanez** [comment]
@jansenbe That one is funny, as it's considering the {12} {0}, and {whatever} as a Token. So, the ObjectField, first is parsing the "valid" tokens:

```c#
var fieldXml = parser.ParseString(templateFieldElement.ToString(), "~sitecollection", "~site");
```
then, is calling the IsFieldXmlValid function, which is calling:
```c#
var leftOverTokens = parser.GetLeftOverTokens(fieldXml);
```
and that GetLeftOverTokens function, is using a regular expression to get all the possible GUIDs and Tokens, and it validates that GUIDs are valid, and returns a list with the tokens that are not GUIDs. In the case of the issue, it returns a list with {12} and {0} items.
```c#
var matches = Regex.Matches(input, "(?<guid>\\{\\S{8}-\\S{4}-\\S{4}-\\S{4}-\\S{12}?\\})|(?<token>\\{.+?\\})").OfType<Match>().Select(m => m.Value);
```
then, back to the IsFieldXmlValid function, if that list is not empty, it considers the field as invalid:
```c#
 else
{
    //Some tokens where not replaced
    isValid = false;
}
```
Why the regular expression in GetLeftOverTokens is also getting Tokens, and not just Guids? I've tested to change the Regular expression to only get Guids, and works OK for this issue, but not sure if it has side effects. As far as I see, everytime we call the IsFieldXmlValid function, we have called first the Token.ParseString function, so all our well-known tokens should be already replaced, and we should only worry about the Guids...
I can do a PullRequest if you think is a valid solution. 
/cc @erwinvanhunen
